lib/portage/gpkg.py

```
@@ -0,0 +1,1564 @@
+			self._get_tar_format_from_stats(
+				*self._check_pre_image_files(root_dir))
+
+		# gpkg container
+		container = tarfile.TarFile(name=self.gpkg_file, mode='w',
+			format=container_tar_format)
+
+		# gpkg version
+		gpkg_version_file = tarfile.TarInfo(self.gpkg_version)
+		gpkg_version_file.mtime = datetime.utcnow().timestamp()
+		container.addfile(gpkg_version_file)
+
+		compression_cmd = self._get_compression_cmd()
+
+		# metadata
+		self._add_metadata(container, metadata, compression_cmd)
+
+		# image
+		if self.create_signature:
+			checksum_info = checksum_helper(self.settings,
+				gpg_operation=checksum_helper.SIGNING)
+		else:
+			checksum_info = checksum_helper(self.settings)
+
+		image_tarinfo = self._create_tarinfo("image")
+		image_tarinfo.mtime = datetime.utcnow().timestamp()
+		with tar_stream_writer(image_tarinfo, container,
+			image_tar_format, compression_cmd, checksum_info) as image_writer:
+			with tarfile.open(mode='w|', fileobj=image_writer,
+				format=image_tar_format) as image_tar:
+				image_tar.add(root_dir, ".", recursive=True)
+
+		image_tarinfo = container.getmember(image_tarinfo.name)
+		self._record_checksum(checksum_info, image_tarinfo)
+
+		if self.create_signature:
+			self._add_signature(checksum_info, image_tarinfo, container)
+
+		self._add_manifest(container)
+		container.close()
+
+	def decompress(self, decompress_dir):
+		"""
+		decompress current gpkg to decompress_dir
+		"""
+		decompress_dir = normalize_path(_unicode_decode(decompress_dir,
+			encoding=_encodings['fs'], errors='strict'))
+
+		self._verify_binpkg()
+
+		with tarfile.open(self.gpkg_file, 'r') as container:
+			image_tarinfo, image_comp = \
+				self._get_inner_tarinfo(container, 'image')
+
+			with tar_stream_reader(container.extractfile(image_tarinfo),
+				self._get_decompression_cmd(image_comp)) as image_tar:
+
+				with tarfile.open(mode='r|', fileobj=image_tar) as image:
+					try:
+						image.extractall(decompress_dir)
+					except:
```

Bare except is bad since it catches BaseException. It's fine to use `except Exception:`, but we should somehow report to the caller that an error has occurred.